.github/workflows/build.yml

```diff
@@ -34,3 +36,60 @@ jobs:
 
       - name: TypeScript build
         run: yarn ts:build
+  deploy:
+    needs: build
+    # only run deploy on pushes and releases
+    if: ${{ github.event_name == push || github.event_name == release }}
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      # set target deployment to production on release and get release tag
+      - name: Configuration for releases
+        if: ${{ github.event_name == release }}
+        run: |
+          echo "DEPLOYMENT_TARGET=production" >> $GITHUB_ENV
+          echo "RELEASE_TAG=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
+
+      # set target deployment to staging on push
+      - name: Configuration for pushes
+        if: ${{ github.event_name == push }}
+        run: echo "DEPLOYMENT_TARGET=staging" >> $GITHUB_ENV
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and push version image
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          cache-to: uwcsclub/codey-bot:cache
+          cache-from: uwcsclub/codey-bot:cache
+          file: docker/${{env.DEPLOYMENT_TARGET}}/Dockerfile
+          push: true
+          tags: ${{ github.event_name == release && format('uwcsclub/codey-bot:{0},uwcsclub/codey-bot:{1}' , env.RELEASE_TAG, github.sha) || format('uwcsclub/codey-bot:{0}', github.sha)}}
```

Comment:
Exactly how did you want the tag(s) to look like?

Reply:
the idea is to have
`uwcsclub/codey-bot:${{ env.RELEASE_TAG }},uwcsclub/codey-bot:${{ github.sha }}`
for releases, and 
`uwcsclub/codey-bot:${{ github.sha }}`
for pushes

Reply:
simplified it a bit
for additional context, github workflows doesn't seem to support ternary; this is the existing workaround on a as of yet unclosed ticket
https://github.com/actions/runner/issues/409